// dataUser/main.mjs
export default function (data = { name: '', dateBirth: '', purpose: '' }) {
  const fullname = data.name.split(' ');
  const toCapitalize = (str) => str[0].toUpperCase() + str.slice(1).toLowerCase();

  const getRandomID = () => Math.random().toString(16).substring(2, 12);
  const setFirstName = () => toCapitalize(fullname[0]);
  const setLastName = () => toCapitalize(fullname[1])
  const setAge = () => {
    const [userDay,  userMonth, userYear] = data.dateBirth.split('.')
    const correctUserMonth = userMonth - 1;
    const today = new Date();
    const todayYear = today.getFullYear();
    const todayMonth = today.getMonth();
    const todayDay = today.getDate();
    let age = todayYear - userYear;

    if (correctUserMonth > todayMonth) {
      age -= 1;
    } else if (correctUserMonth === todayMonth && userDay > todayDay) {
      age -= 1;
    }
    return age;
  };

  return ({
    id: getRandomID(),
    firstName: setFirstName(),
    lastName: setLastName(),
    dateBirth: data.dateBirth,
    age: setAge(),
    purpose: toCapitalize(data.purpose)
  })
}
